toolkit/scripts/toolchain.mk

```diff
@@ -139,14 +145,49 @@ $(raw_toolchain): $(toolchain_files)
 			$(SPECS_DIR) \
 			$(SOURCE_URL)
 
-# Always start with a fresh toolchain chroot when rebuilding toolchain RPMs
+# This target establishes a cache of toolchain RPMs for partially rehydrating the toolchain from package repos.
+# $(toolchain_from_repos) is a staging folder for these RPMs. We use the toolchain manifest to get a list of
+# filenames for each possibly-rehydrated RPM. We attempt to download each RPM from $(PACKAGE_URL_LIST). In cases
+# where an RPM is unavailable/download fails, we create an empty file. The build_official_toolchain_rpms.sh script
+# knows the distinction between empty/non-empty files.
 #
+# When is it meaningful to attempt to partially rehydrate? It makes no sense to partially rehydrate if:
+# - REBUILD_TOOLCAHIN = n: We aren't building the toolchain, so we defer to the full rehydration step
+# - INCREMENTAL_TOOLCHAIN = n: We explicitly want to build a full toolchain
+# - ALLOW_TOOLCHAIN_DOWNLOAD_FAIL = n: This flag explicitly disables partial toolchain rehydration from repos
```

Comment:
Can you add some comment as to why you would want to use this?

Reply:
Done, thanks!